### PR TITLE
chore: Upgrade to Angular to 2.4.8

### PR DESCRIPTION
**Pull request by mcacek**
Upgrade to Angular 2.4.8, angular-cli to latest, and upgrade to the latest release of angular-material. Also fix the bulk of linting errors.

### CODE REVIEW

File: ng-client/package.json [hunk 1]
```diff
@@ -13,16 +13,16 @@
...
-    "@angular/material": "2.0.0-beta.1",
-    "@angular/platform-browser": "2.3.1",
-    "@angular/platform-browser-dynamic": "2.3.1",
-    "@angular/router": "3.3.1",
-    "angular2-jwt": "0.1.27",
+    "@angular/common": "2.4.8",
+    "@angular/core": "2.4.8",
+    "@angular/forms": "2.4.8",
+    "@angular/http": "2.4.8",
+    "@angular/material": "^2.0.0-beta.2",
```
Comment (jondejong) [inline ng-client/package.json]: The carrot is dangerous.

File: ng-client/src/app/employee/employee-new/employee-new.component.html [hunk 2]
```diff
@@ -2,41 +2,39 @@
...
+                    [(ngModel)]="employee.lastName"
+                    name="lastName"
+                    dividerColor="warm"
+                    #lastName="ngModel">
+        </md-input-container>
         <div [hidden]="lastName.valid || lastName.pristine" class="alert">
           Last Name is required
         </div>
 
-        <!--Email-->
```
Comment (jondejong) [inline ng-client/src/app/employee/employee-new/employee-new.component.html]: Why remove the comments of some of the fields, but not others?

**Comment by mcacek** [comment]
Yeeaah it is. I will remove it.

File: ng-client/src/app/employee/employee-new/employee-new.component.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { Employee } from '../Employee';
 @Component({
   selector: 'app-employee-new',
   templateUrl: './employee-new.component.html',
-  styleUrls: ['././employee-new.component.scss']
+  styleUrls: ['././employee-new.component.scss'],
```
Comment (jondejong) [inline ng-client/src/app/employee/employee-new/employee-new.component.ts]: One extra './'?

**Comment by mcacek** [comment]
I think I just missed them.

**Pull request merged**